src/main/java/org/quiltmc/loader/impl/discovery/ModResolver.java

```diff
@@ -1611,8 +1612,14 @@ public ModDep(Logger logger, ModLoadOption source, ModDependency publicDep, ModI
 
 		@Override
 		ModDep put(DependencyHelper<LoadOption, ModLink> helper) throws ContradictionException {
-			List<LoadOption> clause = new ArrayList<>();
-			clause.addAll(validOptions);
+			List<LoadOption> clause = new ArrayList<>(validOptions.size());
```

Comment:
Should be:
```suggestion
			List<LoadOption> clause = new ArrayList<>(validOptions.size() + 1);
```
Since we add a negated option too. Although if you're trying to optimise then it would be better to replace `ArrayList<LoadOption>` with `LoadOption[]` and use an index. But that's really not necessary.

Reply:
Resolved with array